catalyst-gateway/bin/src/db/index/schema/mod.rs

```diff
@@ -162,21 +163,25 @@ fn generate_cql_schema_version() -> String {
 }
 
 /// Get the namespace for a particular db configuration
-pub(crate) fn namespace(cfg: &cassandra_db::EnvVars) -> String {
+pub(crate) fn namespace(cfg: &cassandra_db::EnvVars, network: Network) -> anyhow::Result<String> {
     // Build and set the Keyspace to use.
-    format!(
-        "{}_{}",
-        cfg.namespace.as_str(),
+    let namespace = match cfg.namespace.as_str() {
+        "persistent" => "p",
+        "volatile" => "v",
+        other => anyhow::bail!("Invalid namespace: {other}"),
+    };
```

Comment:
Maybe it would be better to move this match somewhere to the initialization? Otherwise this code would be potentially endlessly executed in the `retry_init` loop.

Reply:
Thanks, you're right.